### PR TITLE
Pr trigger

### LINKED ISSUE #23: PR validation
How should PR validation / PR triggers be handled?

### PR DESCRIPTION
**Pull request by simongottschlag**
Fixes #23

### CODE REVIEW

**Comment by simongottschlag** [comment]
@ulrikstrid @phillebaba this will make it possible to run the CI as validation for a pull request without triggering the CD.

File: gitops/build/main.yaml [hunk 1]
```diff
@@ -96,7 +100,8 @@ stages:
               DOCKER_BUILD_ARGS: ${{ parameters.dockerBuildArgs }}
               ARTIFACT_FILE_NAME: $(artifactFileName)
             displayName: "Build and save image"
-            condition: and(succeeded(), eq(variables['Build.sourceBranch'], '${{ parameters.sourceBranch }}'))
+            ${{ if not(eq(variables['Build.Reason'], 'PullRequest')) }}:
```
Comment (ulrikstrid) [inline gitops/build/main.yaml:103]: "If this is not a PullRequest, add this condtion" -> We will always build and save image on PR, is this intended?

**Review by ulrikstrid — COMMENTED** [review]
How does the CD trigger work?

File: gitops/build/main.yaml (hunk 1 above)
Comment (simongottschlag) [reply gitops/build/main.yaml:103]: @ulrikstrid Yes, we need to save it to scan it with trivy. But we never store it as an artifact.

Comment (simongottschlag) [reply gitops/build/main.yaml:103]: @ulrikstrid The CD is triggered by the stage "cd_trigger" being completed.

File: gitops/README.md [hunk 2]
```diff
@@ -84,7 +84,9 @@ resources:
   pipelines:
     - pipeline: ciPipeline
       source: ci-podinfo
-      trigger: true
+      trigger:
+        stages:
+        - cd_trigger # Stage will only trigger if the build reason isn't pull request
```
Comment (phillebaba) [inline gitops/README.md:89]: Is it not possible to trigger using branches?
```
# this is being defined in app-ci pipeline
resources:
  pipelines:
  - pipeline: securitylib   # Name of the pipeline resource
    source: security-lib-ci # Name of the pipeline referenced by the pipeline resource
    trigger: 
      branches:
      - releases/*
      - master
```

Comment (simongottschlag) [reply gitops/README.md:89]: @phillebaba yeah, but then we can't be flexible around someone using non-default names on their branches. `develop` etc.

Comment (phillebaba) [reply gitops/README.md:89]: Ẁhy not? Just change master to develop? This is set in the CD pipeline and not in the template.

**Review by ulrikstrid — APPROVED** [review]

**Review by phillebaba — APPROVED** [review]
LGTM

**Pull request merged**